src/compiler/crystal/command/repl.cr

```diff
@@ -34,14 +35,23 @@ class Crystal::Command
     end
 
     if options.empty?
+      welcome
       repl.run
     else
       filename = options.shift
       unless File.file?(filename)
         error "File '#{filename}' doesn't exist"
       end
 
+      welcome
```

Comment:
If you run the interpreter with the idea to pipe the result to some other program, this will prohibit that. I'd only show the welcome message if you don't pass arguments. Just my opinion though.

Reply:
Agree that this is a problem. To mitigate this,

* Print it only on tty similarly to

  https://github.com/crystal-lang/crystal/blob/9577ba726add29734efdd927e35eaeb7d2b8f26f/src/colorize.cr#L140

* Print it to stderr in the first place

Reply:
Ah does it accept a filename as an argument? If so, yes, maybe it's worth removing the message in that case too

Reply:
> Ah does it accept a filename as an argument? If so, yes, maybe it's worth removing the message in that case too

I'm not sure about this last one. If you are testing locally some file, like `crystal i spec/some_spec.cr`, I still want this message to appear by default.

Reply:
I agree. As long as it's printed to STDERR, that should be fine.